routing/single_mwm_router.hpp

```diff
@@ -44,6 +51,7 @@ class SingleMwmRouter
 
   string const m_name;
   Index const & m_index;
+  GetTrafficColoringFn const & m_getTrafficColoringFn;
```

Comment:
1. Let's do interface instead of storing the const ref.
2. As for if it's guaranteed that when we call m_getTrafficColoringFn traffic information is valid (that mean RoutingSession::GetTrafficInfo() is valid).
I'm moving |RoutingSession::m_trafficInfo| this way:
```c++
private:
  map<MwmSet::MwmId, shared_ptr<traffic::TrafficInfo>> m_trafficInfo;
  unique_ptr<AsyncRouter> m_router;
...
```
Now |m_trafficInfo| is before |m_router|
So |m_trafficInfo| is created before |m_router| and is destroyed after |m_router|.

Now it looks like everything's ok. |RoutingSession::GetTrafficInfo()| can be called from |m_router|. Before destroying |m_router| calls join for routing thread. But when |m_router| is alive |m_trafficInfo| is alive.